tests/src/test/scala/system/basic/WskRuleTests.scala

```diff
@@ -102,21 +103,28 @@ abstract class WskRuleTests extends TestHelpers with WskTestHelpers {
       case (trigger, status) =>
         if (status == active) wsk.rule.enable(ruleName) else wsk.rule.disable(ruleName)
 
-        // Needs to be retried since the enable/disable causes a cache invalidation which needs to propagate first
-        retry(
-          {
-            wsk.rule
-              .create(ruleName, trigger, actionName, update = true)
-              .stdout
-              .parseJson
-              .asJsObject
-              .fields
-              .get("status") shouldBe status
-
-            wsk.rule.get(ruleName).stdout.parseJson.asJsObject.fields.get("status") shouldBe status
-          },
-          10,
-          Some(1.second))
+        if (cli) {
```

Comment:
As part of future work, the base tests shouldn't know or care whether the the REST or CLI is being used. We should be able to provide a special function to strip out the CLI header that have the same interface that the REST tests use to prevent adding this special logic in the base tests. Maybe we can override `RunResult` for the CLI tests that has a method that removes the header.

Reply:
As part of future work, the base tests shouldn't know or care whether the the REST or CLI is being used. We should be able to provide a special function to strip out the CLI header that have the same interface that the REST tests use to prevent adding this special logic in the base tests. Maybe we can override RunResult for the CLI tests that has a method that removes the header.

Reply:
Also, as part of future work, we should be able to move `Wsk.scala` to incubator-openwhisk-cli.

Reply:
would you like to add that to the issue @dubee ? or maybe create a new one? thankfully `wsk.scala` isn't being duplicated at the moment, so it may be more appropriate as a lower priority down the road

Reply:
example cli header removal method
https://github.com/apache/incubator-openwhisk/blob/f64722498bc2f5eadc356b3ae26ebb85de1bcdbf/tests/src/test/scala/common/WskTestHelpers.scala#L278